static/js/message_view_header.js

```diff
@@ -73,7 +73,12 @@ export function colorize_message_view_header() {
     if (filter === undefined || !filter._sub) {
         return;
     }
-    $("#message_view_header .stream > .fa").css("color", filter._sub.color);
+    let sel = "#message_view_header .stream > .fa";
+    if (filter._sub.is_web_public) {
+        // web public streams have a custom zulip globe icon
+        sel = "#message_view_header .stream > .zulip-icon";
+    }
```

Comment:
Could we just use `#message_view_header a.stream > i` as a fixed selector, to get the icon element in the stream name block?